src/lightning_fabric/strategies/strategy.py

```diff
@@ -247,23 +247,36 @@ def get_optimizer_state(self, optimizer: Optimizer) -> Dict[str, Tensor]:
         # for optimizers that are not sharded, we return the state dict on all ranks
         return optimizer.state_dict()
 
-    def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
-        torch.cuda.empty_cache()
-        return self.checkpoint_io.load_checkpoint(checkpoint_path)
+    def load_checkpoint(
+        self, path: _PATH, state: Optional[Dict[str, Union[Module, Optimizer, Any]]] = None
+    ) -> Dict[str, Any]:
+        """Load the contents from a checkpoint and restore the state of the given objects.
 
-    def load_module_state_dict(self, module: Module, checkpoint: Mapping[str, Any]) -> None:
-        # TODO(fabric): Integrate this into Lightning Fabric
-        module.load_state_dict(checkpoint["state_dict"])
+        Args:
+            path: A path to where the file is located
+            state: A dictionary of objects whose state will be restored in-place from the checkpoint path.
+                If no state is given, then the checkpoint will be returned in full.
 
-    def load_optimizer_state_dict(
-        self, optimizers: Union[Optimizer, Iterable[Optimizer]], checkpoint: Mapping[str, Any]
-    ) -> None:
-        if not isinstance(optimizers, Iterable):
-            optimizers = [optimizers]
-        optimizer_states = checkpoint["optimizer_states"]
-        for optimizer, opt_state in zip(optimizers, optimizer_states):
-            optimizer.load_state_dict(opt_state)
-            _optimizer_to_device(optimizer, self.root_device)
+        Returns:
+            The remaining items that were not restored into the given state dictionary. If no state dictionary is
+            given, the full checkpoint will be returned.
+        """
+        torch.cuda.empty_cache()
+        checkpoint = self.checkpoint_io.load_checkpoint(path)
+        if not state:
+            return checkpoint
+
+        for name, obj in state.copy().items():
```

Comment:
Why this `copy()`?

Reply:
Because I'm modifying the keys as I am iterating over the dict.  I believe it is valid use here, do you see a problem?

`state[name] = checkpoint.pop(name)`